build.gradle

```diff
@@ -271,6 +282,40 @@ project(':sbe-tool') {
 
     compileTestJava.dependsOn compileGeneratedJava
 
+    testing {
+        suites {
+            test {
+                dependencies {
+                    implementation files('build/classes/java/generated')
+                    implementation "org.hamcrest:hamcrest:${hamcrestVersion}"
+                    implementation "org.mockito:mockito-core:${mockitoVersion}"
+                    implementation "org.junit.jupiter:junit-jupiter-params:${junitVersion}"
```

Comment:
Note that we're using `jvm-test-suites` to add new source sets in a Gradle 9 compatible manner.
Here `implementation` means `testImplementation` in old money.

Reply:
<details><summary>Proof</summary>
<p>
```
 zach@worky-ii  ~/src/real-logic/simple-binary-encoding   feature/dtos ⍟6  ↵ 1  ./gradlew sbe-tool:dependencies

------------------------------------------------------------
Project ':sbe-tool'
------------------------------------------------------------

annotationProcessor - Annotation processors and their dependencies for source set 'main'.
No dependencies

api - API dependencies for source set 'main'. (n)
\--- org.agrona:agrona:[1.19.2,2.0[ (n)

apiElements - API elements for main. (n)
No dependencies

checkstyle - The Checkstyle libraries to be used for this project.
\--- com.puppycrawl.tools:checkstyle:9.3
     +--- info.picocli:picocli:4.6.2
     +--- org.antlr:antlr4-runtime:4.9.3
     +--- commons-beanutils:commons-beanutils:1.9.4
     |    \--- commons-collections:commons-collections:3.2.2
     +--- com.google.guava:guava:31.0.1-jre
     |    +--- com.google.guava:failureaccess:1.0.1
     |    +--- com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava
     |    +--- com.google.code.findbugs:jsr305:3.0.2
     |    +--- org.checkerframework:checker-qual:3.12.0
     |    +--- com.google.errorprone:error_prone_annotations:2.7.1
     |    \--- com.google.j2objc:j2objc-annotations:1.3
     +--- org.reflections:reflections:0.10.2
     |    +--- org.javassist:javassist:3.28.0-GA
     |    \--- com.google.code.findbugs:jsr305:3.0.2
     \--- net.sf.saxon:Saxon-HE:10.6

compileClasspath - Compile classpath for source set 'main'.
\--- org.agrona:agrona:{strictly [1.19.2,2.0[; prefer 1.19.2} -> 1.19.2

compileOnly - Compile only dependencies for source set 'main'. (n)
No dependencies

compileOnlyApi - Compile only API dependencies for source set 'main'. (n)
No dependencies

default - Configuration for default artifacts. (n)
No dependencies

generatedAnnotationProcessor - Annotation processors and their dependencies for source set 'generated'.
No dependencies

generatedCompileClasspath - Compile classpath for source set 'generated'.
No dependencies

generatedCompileOnly - Compile only dependencies for source set 'generated'. (n)
No dependencies

generatedImplementation - Implementation only dependencies for source set 'generated'. (n)
No dependencies

generatedRuntimeClasspath - Runtime classpath of source set 'generated'.
No dependencies

generatedRuntimeOnly - Runtime only dependencies for source set 'generated'. (n)
No dependencies

implementation - Implementation only dependencies for source set 'main'. (n)
No dependencies

javadocElements - javadoc elements for main. (n)
No dependencies

mainSourceElements - List of source directories contained in the Main SourceSet. (n)
No dependencies

propertyTestAnnotationProcessor - Annotation processors and their dependencies for source set 'property test'.
No dependencies

propertyTestCompileClasspath - Compile classpath for source set 'property test'.
+--- project :sbe-tool (*)
+--- net.jqwik:jqwik:1.8.0
|    +--- org.apiguardian:apiguardian-api:1.1.2
|    +--- net.jqwik:jqwik-api:1.8.0
|    |    +--- org.apiguardian:apiguardian-api:1.1.2
|    |    +--- org.opentest4j:opentest4j:1.3.0
|    |    \--- org.junit.platform:junit-platform-commons:1.10.0
|    |         +--- org.junit:junit-bom:5.10.0
|    |         |    +--- org.junit.jupiter:junit-jupiter:5.10.0 (c)
|    |         |    +--- org.junit.jupiter:junit-jupiter-api:5.10.0 (c)
|    |         |    +--- org.junit.jupiter:junit-jupiter-params:5.10.0 (c)
|    |         |    \--- org.junit.platform:junit-platform-commons:1.10.0 (c)
|    |         \--- org.apiguardian:apiguardian-api:1.1.2
|    +--- net.jqwik:jqwik-web:1.8.0
|    |    +--- org.apiguardian:apiguardian-api:1.1.2
|    |    +--- net.jqwik:jqwik-api:1.8.0 (*)
|    |    \--- org.opentest4j:opentest4j:1.3.0
|    \--- net.jqwik:jqwik-time:1.8.0
|         +--- org.apiguardian:apiguardian-api:1.1.2
|         +--- net.jqwik:jqwik-api:1.8.0 (*)
|         \--- org.opentest4j:opentest4j:1.3.0
+--- org.json:json:20230618
+--- org.junit.jupiter:junit-jupiter:5.10.0
|    +--- org.junit:junit-bom:5.10.0 (*)
|    +--- org.junit.jupiter:junit-jupiter-api:5.10.0
|    |    +--- org.junit:junit-bom:5.10.0 (*)
|    |    +--- org.opentest4j:opentest4j:1.3.0
|    |    +--- org.junit.platform:junit-platform-commons:1.10.0 (*)
|    |    \--- org.apiguardian:apiguardian-api:1.1.2
|    \--- org.junit.jupiter:junit-jupiter-params:5.10.0
|         +--- org.junit:junit-bom:5.10.0 (*)
|         +--- org.junit.jupiter:junit-jupiter-api:5.10.0 (*)
|         \--- org.apiguardian:apiguardian-api:1.1.2
\--- org.agrona:agrona:{strictly [1.19.2,2.0[; prefer 1.19.2} -> 1.19.2

propertyTestCompileOnly - Compile only dependencies for source set 'property test'. (n)
No dependencies

propertyTestImplementation - Implementation only dependencies for source set 'property test'. (n)
+--- project sbe-tool (n)
+--- net.jqwik:jqwik:1.8.0 (n)
\--- org.json:json:20230618 (n)

propertyTestRuntimeClasspath - Runtime classpath of source set 'property test'.
+--- project :sbe-tool (*)
+--- net.jqwik:jqwik:1.8.0
|    +--- org.apiguardian:apiguardian-api:1.1.2
|    +--- net.jqwik:jqwik-api:1.8.0
|    |    +--- org.apiguardian:apiguardian-api:1.1.2
|    |    +--- org.opentest4j:opentest4j:1.3.0
|    |    \--- org.junit.platform:junit-platform-commons:1.10.0
|    |         \--- org.junit:junit-bom:5.10.0
|    |              +--- org.junit.jupiter:junit-jupiter:5.10.0 (c)
|    |              +--- org.junit.jupiter:junit-jupiter-api:5.10.0 (c)
|    |              +--- org.junit.jupiter:junit-jupiter-engine:5.10.0 (c)
|    |              +--- org.junit.jupiter:junit-jupiter-params:5.10.0 (c)
|    |              +--- org.junit.platform:junit-platform-commons:1.10.0 (c)
|    |              +--- org.junit.platform:junit-platform-engine:1.10.0 (c)
|    |              \--- org.junit.platform:junit-platform-launcher:1.10.0 (c)
|    +--- net.jqwik:jqwik-web:1.8.0
|    |    +--- org.apiguardian:apiguardian-api:1.1.2
|    |    +--- net.jqwik:jqwik-api:1.8.0 (*)
|    |    \--- org.opentest4j:opentest4j:1.3.0
|    +--- net.jqwik:jqwik-time:1.8.0
|    |    +--- org.apiguardian:apiguardian-api:1.1.2
|    |    +--- net.jqwik:jqwik-api:1.8.0 (*)
|    |    \--- org.opentest4j:opentest4j:1.3.0
|    \--- net.jqwik:jqwik-engine:1.8.0
|         +--- org.junit.platform:junit-platform-engine:1.10.0
|         |    +--- org.junit:junit-bom:5.10.0 (*)
|         |    +--- org.opentest4j:opentest4j:1.3.0
|         |    \--- org.junit.platform:junit-platform-commons:1.10.0 (*)
|         +--- org.apiguardian:apiguardian-api:1.1.2
|         +--- net.jqwik:jqwik-api:1.8.0 (*)
|         +--- org.opentest4j:opentest4j:1.3.0
|         \--- org.junit.platform:junit-platform-commons:1.10.0 (*)
+--- org.json:json:20230618
+--- org.junit.jupiter:junit-jupiter:5.10.0
|    +--- org.junit:junit-bom:5.10.0 (*)
|    +--- org.junit.jupiter:junit-jupiter-api:5.10.0
|    |    +--- org.junit:junit-bom:5.10.0 (*)
|    |    +--- org.opentest4j:opentest4j:1.3.0
|    |    \--- org.junit.platform:junit-platform-commons:1.10.0 (*)
|    +--- org.junit.jupiter:junit-jupiter-params:5.10.0
|    |    +--- org.junit:junit-bom:5.10.0 (*)
|    |    \--- org.junit.jupiter:junit-jupiter-api:5.10.0 (*)
|    \--- org.junit.jupiter:junit-jupiter-engine:5.10.0
|         +--- org.junit:junit-bom:5.10.0 (*)
|         +--- org.junit.platform:junit-platform-engine:1.10.0 (*)
|         \--- org.junit.jupiter:junit-jupiter-api:5.10.0 (*)
+--- org.junit.platform:junit-platform-launcher -> 1.10.0
|    +--- org.junit:junit-bom:5.10.0 (*)
|    \--- org.junit.platform:junit-platform-engine:1.10.0 (*)
\--- org.agrona:agrona:{strictly [1.19.2,2.0[; prefer 1.19.2} -> 1.19.2

propertyTestRuntimeOnly - Runtime only dependencies for source set 'property test'. (n)
No dependencies

runtimeClasspath - Runtime classpath of source set 'main'.
\--- org.agrona:agrona:{strictly [1.19.2,2.0[; prefer 1.19.2} -> 1.19.2

runtimeElements - Elements of runtime for main. (n)
No dependencies

runtimeOnly - Runtime only dependencies for source set 'main'. (n)
No dependencies

signatures (n)
No dependencies

sourcesElements - sources elements for main. (n)
No dependencies

testAnnotationProcessor - Annotation processors and their dependencies for source set 'test'.
No dependencies

testCompileClasspath - Compile classpath for source set 'test'.
+--- org.agrona:agrona:{strictly [1.19.2,2.0[; prefer 1.19.2} -> 1.19.2
+--- org.hamcrest:hamcrest:2.2
+--- org.mockito:mockito-core:4.11.0
|    +--- net.bytebuddy:byte-buddy:1.12.19
|    \--- net.bytebuddy:byte-buddy-agent:1.12.19
+--- org.junit.jupiter:junit-jupiter-params:5.10.0
|    +--- org.junit:junit-bom:5.10.0
|    |    +--- org.junit.jupiter:junit-jupiter:5.10.0 (c)
|    |    +--- org.junit.jupiter:junit-jupiter-api:5.10.0 (c)
|    |    +--- org.junit.jupiter:junit-jupiter-params:5.10.0 (c)
|    |    \--- org.junit.platform:junit-platform-commons:1.10.0 (c)
|    +--- org.junit.jupiter:junit-jupiter-api:5.10.0
|    |    +--- org.junit:junit-bom:5.10.0 (*)
|    |    +--- org.opentest4j:opentest4j:1.3.0
|    |    +--- org.junit.platform:junit-platform-commons:1.10.0
|    |    |    +--- org.junit:junit-bom:5.10.0 (*)
|    |    |    \--- org.apiguardian:apiguardian-api:1.1.2
|    |    \--- org.apiguardian:apiguardian-api:1.1.2
|    \--- org.apiguardian:apiguardian-api:1.1.2
\--- org.junit.jupiter:junit-jupiter:5.10.0
     +--- org.junit:junit-bom:5.10.0 (*)
     +--- org.junit.jupiter:junit-jupiter-api:5.10.0 (*)
     \--- org.junit.jupiter:junit-jupiter-params:5.10.0 (*)

testCompileOnly - Compile only dependencies for source set 'test'. (n)
No dependencies

testImplementation - Implementation only dependencies for source set 'test'. (n)
+--- unspecified (n)
+--- org.hamcrest:hamcrest:2.2 (n)
+--- org.mockito:mockito-core:4.11.0 (n)
\--- org.junit.jupiter:junit-jupiter-params:5.10.0 (n)

testRuntimeClasspath - Runtime classpath of source set 'test'.
+--- org.agrona:agrona:{strictly [1.19.2,2.0[; prefer 1.19.2} -> 1.19.2
+--- org.hamcrest:hamcrest:2.2
+--- org.mockito:mockito-core:4.11.0
|    +--- net.bytebuddy:byte-buddy:1.12.19
|    +--- net.bytebuddy:byte-buddy-agent:1.12.19
|    \--- org.objenesis:objenesis:3.3
+--- org.junit.jupiter:junit-jupiter-params:5.10.0
|    +--- org.junit:junit-bom:5.10.0
|    |    +--- org.junit.jupiter:junit-jupiter:5.10.0 (c)
|    |    +--- org.junit.jupiter:junit-jupiter-api:5.10.0 (c)
|    |    +--- org.junit.jupiter:junit-jupiter-engine:5.10.0 (c)
|    |    +--- org.junit.jupiter:junit-jupiter-params:5.10.0 (c)
|    |    +--- org.junit.platform:junit-platform-launcher:1.10.0 (c)
|    |    +--- org.junit.platform:junit-platform-commons:1.10.0 (c)
|    |    \--- org.junit.platform:junit-platform-engine:1.10.0 (c)
|    \--- org.junit.jupiter:junit-jupiter-api:5.10.0
|         +--- org.junit:junit-bom:5.10.0 (*)
|         +--- org.opentest4j:opentest4j:1.3.0
|         \--- org.junit.platform:junit-platform-commons:1.10.0
|              \--- org.junit:junit-bom:5.10.0 (*)
+--- org.junit.jupiter:junit-jupiter:5.10.0
|    +--- org.junit:junit-bom:5.10.0 (*)
|    +--- org.junit.jupiter:junit-jupiter-api:5.10.0 (*)
|    +--- org.junit.jupiter:junit-jupiter-params:5.10.0 (*)
|    \--- org.junit.jupiter:junit-jupiter-engine:5.10.0
|         +--- org.junit:junit-bom:5.10.0 (*)
|         +--- org.junit.platform:junit-platform-engine:1.10.0
|         |    +--- org.junit:junit-bom:5.10.0 (*)
|         |    +--- org.opentest4j:opentest4j:1.3.0
|         |    \--- org.junit.platform:junit-platform-commons:1.10.0 (*)
|         \--- org.junit.jupiter:junit-jupiter-api:5.10.0 (*)
\--- org.junit.platform:junit-platform-launcher -> 1.10.0
     +--- org.junit:junit-bom:5.10.0 (*)
     \--- org.junit.platform:junit-platform-engine:1.10.0 (*)

testRuntimeOnly - Runtime only dependencies for source set 'test'. (n)
No dependencies

(c) - A dependency constraint, not a dependency. The dependency affected by the constraint occurs elsewhere in the tree.
(*) - Indicates repeated occurrences of a transitive dependency subtree. Gradle expands transitive dependency subtrees only once per project; repeat occurrences only display the root of the subtree, followed by this annotation.

(n) - A dependency or dependency configuration that cannot be resolved.

A web-based, searchable dependency report is available by adding the --scan option.

```
</p>
</details>